src/Admin/DownloadPaths/class-dlm-downloads-path-helper.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+} // Exit if accessed directly
+
+/**
+ * DLM_Downloads_Path_Helper class.
+ *
+ * @since 5.0.0
+ */
+class DLM_Downloads_Path_Helper {
+
+	/**
+	 * Saves the download paths if not already existing either for single site or multisite setup.
+	 *
+	 * @param  string  $path  string of download path.
+	 *
+	 * @since 5.0.0
+	 */
+	public static function save_unique_path( $path ) {
+		$saved_paths = self::get_all_paths();
+		$add_file    = true;
+
+		foreach ( $saved_paths as $save_path ) {
+			if ( $path == $save_path['path_val'] ) {
+				$add_file = false;
+				break;
+			}
+		}
+
+		if ( $add_file ) {
+			$lastkey       = array_key_last( $saved_paths );
+			$saved_paths[] = array(
+				'id'       => absint( $saved_paths[ $lastkey ]['id'] ) + 1,
+				'path_val' => trailingslashit( $path ),
+				'enabled'  => true,
+			);
+			// Only allow network admin to add paths
+			if ( ! is_multisite() ) {
+				update_option( 'dlm_allowed_paths', $saved_paths );
+			}
+		}
+	}
+
+	/**
+	 * Saves the download paths either for single site or multisite setup.
+	 * In case of multisite, the blog should be switched to the desired blog before calling this function.
+	 *
+	 * @param  array  $paths  Array of download paths.
+	 *
+	 * @since 5.0.0
+	 */
+	public static function save_paths( $paths ) {
+		update_option( 'dlm_allowed_paths', $paths );
+	}
+
+	/**
+	 * Retrieves all download paths either for single site or multisite setup.
+	 * In case of multisite, the blog should be switched to the desired blog before calling this function.
+	 *
+	 * @return array Array of download paths.
+	 * @since 5.0.0
+	 */
+	public static function get_all_paths() {
+		if ( is_multisite() ) {
+			return get_option( 'dlm_allowed_paths' );
+		} else {
+			$option = get_option( 'dlm_allowed_paths' );
+			// Check if it's string & do combatiblility for < 5.0.0
+			if ( is_string( $option ) ) {
+				if ( '' != $option ) {
+					// Not empty string, save as new format since 5.0.0
+					$paths = array(
+						array(
+							'id'       => 1,
+							'path_val' => $option,
+							'enabled'  => true,
+						),
+					);
+					self::save_paths( $paths );
+
+					return $paths;
+				}
+
+				return array();
+			}
+
+			return $option;
+		}
```

Comment:
nu ai nevoie de else 
```suggestion
			$option = get_option( 'dlm_allowed_paths' );
			// Check if it's string & do combatiblility for < 5.0.0
			if ( is_string( $option ) ) {
				if ( '' != $option ) {
					// Not empty string, save as new format since 5.0.0
					$paths = array(
						array(
							'id'       => 1,
							'path_val' => $option,
							'enabled'  => true,
						),
					);
					self::save_paths( $paths );

					return $paths;
				}

				return array();
			}

			return $option;
		
```